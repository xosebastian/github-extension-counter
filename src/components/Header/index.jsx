import * as React from 'react';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';

export default function Header() {
  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" component="div" sx={{ flexGrow: 0 }}>
            Github Extensions Counter
          </Typography>
        </Toolbar>
      </AppBar>
    </Box>
  );
}
