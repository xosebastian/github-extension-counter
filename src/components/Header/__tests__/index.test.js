/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import Header from '../index';

const getRenderedComponent = () => render(<Header />);

describe('<Header/>', () => {
  it('Must render the component', () => {
    const { baseElement } = getRenderedComponent();
    expect(baseElement).toMatchSnapshot();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    getRenderedComponent();
    expect(spy).not.toHaveBeenCalled();
  });
});
