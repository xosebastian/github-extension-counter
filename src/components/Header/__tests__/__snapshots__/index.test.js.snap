// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Must render the component 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-i9gxme"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 css-7fum8j-MuiTypography-root"
          >
            Github Extensions Counter
          </div>
        </div>
      </header>
    </div>
  </div>
</body>
`;
